### PR TITLE
feat: add details to Analytics migration guide

### PR DESCRIPTION
**Pull request by erinleigh90**
#### Description of changes:

- Adds input paramter and type changes.
- Adds descriptions of what has changed and why
- Organizes content by Provider

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [x] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [x] JS
- [ ] Swift
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/pages/[platform]/build-a-backend/more-features/analytics/analytics-migration-guide/index.mdx [hunk 1]
```diff
@@ -26,110 +26,492 @@ export function getStaticProps(context) {
...
-+   }
-+ });
-
-```
-Notice that in v5, the `Analytics.record` API takes in a second parameter with the string value `AmazonPersonalize` whereas in v6, the `record` function is imported from the `aws-amplify/analytics/personalize` subpath and takes only one parameter. 
+This guide will help you migrate Amplify JavaScript v5's Analytics APIs to the new v6's Analytics APIs.
+
+As of v6 of Amplify, you will now import APIs directly from the `aws-amplify/analytics` path when using the Pinpoint provider. If you are using the Personalize, Kinesis, or Kinesis (Stream) provider, you will need to import separate APIs from the appropriate subpath (ie `import { record } from 'aws-amplify/analytics/personalize'` when using the `record` API with the `AmazonPersonalize` provider) instead of passing in the `provider` or `options.provider` parameter.
+
+<Callout>TODO: The `startSession` API has been removed in v6. (Due to auto-track capabilities?)</Callout>
```
Comment (renebrandel) [inline src/pages/[platform]/build-a-backend/more-features/analytics/analytics-migration-guide/index.mdx:33]: What's the status on this?

Comment (erinleigh90) [reply src/pages/[platform]/build-a-backend/more-features/analytics/analytics-migration-guide/index.mdx:33]: Waiting on some team input here, will update when I get some clarification.

File: src/pages/[platform]/build-a-backend/more-features/analytics/analytics-migration-guide/index.mdx [hunk 2]
```diff
@@ -26,110 +26,492 @@ export function getStaticProps(context) {
...
+
+## Enable and Disable
+
+In both v5 and v6, Analytics is enabled by default. You can disable and enable all Analytics providers using the `enable` and `disable` APIs exported from the `aws-amplify/analytics` path.
```
Comment (cshfang) [inline src/pages/[platform]/build-a-backend/more-features/analytics/analytics-migration-guide/index.mdx:37]: Should Analytics be treated as a singular named category or a plural noun here? I am genuinely uncertain

File: src/pages/[platform]/build-a-backend/more-features/analytics/analytics-migration-guide/index.mdx [hunk 3]
```diff
@@ -26,110 +26,492 @@ export function getStaticProps(context) {
...
+
+## Pinpoint
+
+In v6, `provider` is determined by import path. To use Pinpoint APIs you will import them directly from `aws-amplify/analytics`. These APIs no longer accept parameters specific to Kinesis or Personalize.
+
+### Analytics.record
+
+The `record` API in v6 has not changed in behavior, however, there are a couple of changes to be aware of in v6:
+
+- The `record` API is now synchronous and does not return as events are always buffered.
```
Comment (cshfang) [inline src/pages/[platform]/build-a-backend/more-features/analytics/analytics-migration-guide/index.mdx:72]: Does not return what?

File: src/pages/[platform]/build-a-backend/more-features/analytics/analytics-migration-guide/index.mdx [hunk 4]
```diff
@@ -26,110 +26,492 @@ export function getStaticProps(context) {
...
+- The `immediate` option was removed from the `record` API in v6. Alternatively in v6 you can use the [flushEvents API](/[platform]/build-a-backend/more-features/analytics/record-events/#flush-events) to clear all events from the buffer before calling `record`.
```
Comment (cshfang) [inline src/pages/[platform]/build-a-backend/more-features/analytics/analytics-migration-guide/index.mdx:73]:
```suggestion
- The `immediate` option was removed from the `record` API in v6. Instead, you can now use the [flushEvents API](/[platform]/build-a-backend/more-features/analytics/record-events/#flush-events) to clear all events from the buffer before calling `record`.
```

**Review by cshfang — DISMISSED** [review]

File: src/pages/[platform]/build-a-backend/more-features/analytics/analytics-migration-guide/index.mdx (hunk 3 above)
Comment (erinleigh90) [reply src/pages/[platform]/build-a-backend/more-features/analytics/analytics-migration-guide/index.mdx:70]: Promise<unknown>, I'll try to clarify in the verbiage.

File: src/pages/[platform]/build-a-backend/more-features/analytics/analytics-migration-guide/index.mdx (hunk 2 above)
Comment (erinleigh90) [reply src/pages/[platform]/build-a-backend/more-features/analytics/analytics-migration-guide/index.mdx:35]: I think singular named category (the Analytics category), but it's kind of acting as an adjective/descriptor here.

Comment (erinleigh90) [reply src/pages/[platform]/build-a-backend/more-features/analytics/analytics-migration-guide/index.mdx:35]: I think singular named category would be correct here. I could make it more clear by saying The Analytics category is enabled?

Comment (erinleigh90) [reply src/pages/[platform]/build-a-backend/more-features/analytics/analytics-migration-guide/index.mdx:35]: Google treats it as a singular named category as well.

**Review by renebrandel — APPROVED** [review]

**Review by cshfang — APPROVED** [review]
Lgtm ty

**Review by cshfang — APPROVED** [review]

**Review by renebrandel — APPROVED** [review]

**Pull request merged**